File: leetcode/0169-majority-element/solution.py
class Solution(object):
    def majorityElement(self, nums):
        dic={}
        for i in nums:
            if i in dic:
                dic[i]+=1
            else:
                dic[i]=1
        for key,values in dic.items():
            if values > len(nums)/2:
                return key
        """
        :type nums: List[int]
        :rtype: int
        """

File: leetcode/0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        int i=0;
        while(i<n){
            int j=i;
            while(j<n&&(nums[i]==nums[j])){
                j++;
            }
            if (j-i>n/2){
                return nums[i];
            }
            i=j++;
        }
        return nums[0];
    }
};
